generator/internal/connection_rest_generator.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/connection_rest_generator.h"
+#include "generator/internal/codegen_utils.h"
+#include "generator/internal/descriptor_utils.h"
+#include "generator/internal/predicate_utils.h"
+#include "generator/internal/printer.h"
+#include "absl/memory/memory.h"
+#include <google/protobuf/descriptor.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+ConnectionRestGenerator::ConnectionRestGenerator(
+    google::protobuf::ServiceDescriptor const* service_descriptor,
+    VarsDictionary service_vars,
+    std::map<std::string, VarsDictionary> service_method_vars,
+    google::protobuf::compiler::GeneratorContext* context)
+    : ServiceCodeGenerator("connection_rest_header_path",
+                           "connection_rest_cc_path", service_descriptor,
+                           std::move(service_vars),
+                           std::move(service_method_vars), context) {}
+
+Status ConnectionRestGenerator::GenerateHeader() {
+  HeaderPrint(CopyrightLicenseFileHeader());
+  HeaderPrint(  // clang-format off
```

Comment:
nit: can we copy a file that uses a raw string?

Reply:
Done.